### ISSUE TITLE
Support HEIF and HEIC File formats

### ISSUE THREAD

**Issue by asjones987** [nomacs/nomacs#257]
I may be too early, but i would be great if nomacs supported HEIF and HEIC file formats (some reading shows Apple's HEIC is a bit different than just HEIF for images.

I am sure you have already looked into this and I am confused on the licensing but some things indicate that for free/open source software it may not be an issue vs commercial software.

Really I would like to read and write HEIF/HEIC.  If the standard really took off now that Apple is pushing it (I am not an Apple person) then I would want to batch convert old pictures just to save space  and make them easier to move around.

https://nokiatech.github.io/heif/

https://github.com/nokiatech/heif

https://en.wikipedia.org/wiki/High_Efficiency_Image_File_Format

http://jpgtoheif.com/

.... and it also supports animated images aka like what animated GIF does

**Comment by asjones987** [comment]
Side note I saw that ImageMagick supports the HEEIC format  https://github.com/ImageMagick/ImageMagick

**Comment by diemmarkus** [comment]
we'll try this [Qt plugin](https://github.com/jakar/qt-heif-image-plugin) (since it seems to take longer for Qt to support it [officially](https://bugreports.qt.io/browse/QTBUG-69212)).

**Comment by geroldmeisinger** [comment]
> the next nomacs will be shipped with (rudimentary) heic support.

...but it was not...

https://github.com/nomacs/nomacs/issues/469#issuecomment-2168589297

**Comment by scrubbbbs** [comment]
There was an effort to get it into Qt (at the time) but it stalled. kimageformat-plugins is widespread, or else continue to use qt-heif-image-plugin. We should update the readme and install page [update this page](https://nomacs.org/docs/getting-started/installation/) to reflect that.

**Comment by InfinityAX** [comment]
I checked the install page, and it still needs to be updated. Or, at the very least, instructions could have been more straightforward. For Ubuntu and Linux Mint, run `$sudo apt install kimageformat-plugins qt5-avif-image-plugin`, or you can use the package manager shown below:
![Screenshot from 2024-09-21 10-17-27](https://github.com/user-attachments/assets/adf36e45-3b1d-4737-8190-180347d55f6f)

**Comment by kintaro1981** [comment]
https://github.com/strukturag/libheif
libheif is an HEIF and AVIF file format decoder and encoder.

**Comment by scrubbbbs** [comment]
Since readme is updated this is resolved. kimageformats or another suitable heif plugin for qt is the path forward.

**Comment by step135** [comment]
Still not supported.

**Comment by yuyalun-allen** [comment]
> Still not supported.


I really love this awesome software, but I also encountered the issue discussed in this thread. Even after installing the necessary dependencies (`kimageformats`), `nomacs` still couldn't automatically load HEIF files from a folder—which isn't very intuitive. To resolve this, you need to manually navigate to `Tools -> Add Image Format -> Browse to Load New Image Format`. I hope this helps you and anyone else who comes across this! @step135

**Comment by scrubbbbs** [comment]
> `nomacs` still couldn't automatically load HEIF files from a folder

kimageformats only recognizes ".heif" and ".heic", do you have a different extension?

**Comment by yuyalun-allen** [comment]
> kimageformats only recognizes ".heif" and ".heic", do you have a different extension?


Yes, my Sony camera follows the .HIF suffix convention.  Any better workarounds?

**Comment by yuyalun-allen** [comment]
> `nomacs` still couldn't automatically load HEIF files from a folder

Renaming can indeed solve this problem.

**Comment by scrubbbbs** [comment]
> > kimageformats only recognizes ".heif" and ".heic", do you have a different extension?
> 
> Yes, my Sony camera follows the .HIF suffix convention. Any better workarounds?

We can manually add this extension to nomacs defaults, we do this already for several formats.

**Comment by yuyalun-allen** [comment]
> We can manually add this extension to nomacs defaults, we do this already for several formats.

Thanks for your work! When is this patch expected to be released?

**Comment by scrubbbbs** [comment]
I plan to do a sweep for small fixes like this in the next month or so.

**Comment by scrubbbbs** [comment]
Closing as kimageformats added HIF extension. Not sure when this will drop in the nomacs releases but will keep an eye on it.

In the meantime on Linux if you find HEIC are not opening make sure you have also installed the h.265 codec for heif (Ubuntu/Debian package is `libheif-plugin-libde265`
